std/haxe/macro/Expr.hx

```diff
@@ -169,154 +213,603 @@ enum Unop {
 	OpNegBits;
 }
 
+/**
+	Represents a AST node for the AST.
+	@see http://haxe.org/manual/macro-reification-expression.html
+**/
 typedef Expr = {
+	/**
+		The expression kind.
+	**/
 	var expr : ExprDef;
+	
+	/**
+		The position of the expression.
+	**/
 	var pos : Position;
 }
 
+/**
+	Represents a AST node identical to `Expr`, but it allows constraining the 
+	type of accepted expressions.
+	@see http://haxe.org/manual/macro-ExprOf.html
+**/
 typedef ExprOf<T> = Expr;
 
+/**
+	Represents a case for the AST.
+	@see http://haxe.org/manual/expression-switch.html
+**/
 typedef Case = {
+	/**
+		The value expressions of the case.
+	**/
 	var values : Array<Expr>;
+	/**
+		The guard expressions of the case. Guards restrict cases.
```

Comment:
"Guards restrict cases" is technically correct but sounds kinda strange here. Let's lose it.